test/drenv/envfile_test.py

```diff
@@ -58,9 +57,11 @@
 """
 
 
-def test_valid():
-    f = io.StringIO(valid_yaml)
-    env = envfile.load(f)
+def test_valid(tmpdir):
+    file = tmpdir.join("valid.yaml")
+    file.write(valid_yaml)
+    with open(file) as f:
+        env = envfile.load(f)
```

Comment:
I'm not sure what you try to do in this change and why.

These tests are fine and do not need any change. They check that we can parse
valid yaml using io.StringIO() so we don't need to create a temporary file
(what you do here).

Can you explain this change?